packages/ember-engines/tests/test-helper.js

```diff
@@ -14,7 +28,14 @@ setApplication(Application.create(config.APP));
 setup(QUnit.assert);
 setupSinon();
 
-preloadAssets(manifest).then(start); // This ensures all engine resources are loaded before the tests
+if (macroCondition(!dependencySatisfies('@embroider/core', '*'))) {
+  preloadAssets(manifest).then(start); // This ensures all engine resources are loaded before the tests
```

Comment:
we shouldn't preload assets anymore. Could you please make the tests more real?

Reply:
how do you propose tests be made more real?
deleting these lines makes the tests more real but engines code doesn't work to on the fly load the necessary code where embroider does exactly what it should
that's why for the tests  where it's still needed the assets get preloaded and where it's not we're making the tests better by not preloading them at all

Reply:
@void-mAlex so for this to work, we would need to provide re-exports in the dummy app for each and every service/component/helper/modifier/route/controller/etc.?

sounds like `setupEngine` test helper won't work moving forward

Reply:
setupEngine needs to go the way of the dodo
and to answer your question, quite possibly for testing the components of the engine itself
though you have to pretty much import explicitly all your components in tests in gts format ... so this is the in between step to getting there